src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/TextEditingTargetRenameHelper.java

```diff
@@ -0,0 +1,355 @@
+      ranges_ = new ArrayList<Range>();
+   }
+   
+   public int renameInFile()
+   {
+      init();
+      TextFileType type = editor_.getFileType();
+      if (type.isR() || type.isRhtml() || type.isRmd() || type.isRnw() || type.isRpres())
+         return renameInFileR();
+      return -1;
+   }
+   
+   private int renameInFileR()
+   {
+      Position position = editor_.hasSelection() ?
+            editor_.getSelectionStart() :
+            editor_.getCursorPosition();
+            
+      TokenCursor cursor = editor_.getSession().getMode().getCodeModel().getTokenCursor();
+      if (!cursor.moveToPosition(position, true))
+         return -1;
+      
+      editor_.setCursorPosition(position);
+      
+      // Validate that we're looking at an R identifier (TODO: refactor strings?)
+      String targetValue = cursor.currentValue();
+      String targetType = cursor.currentType();
+      
+      boolean isVariableType =
+            targetType.equals("identifier") ||
+            targetType.equals("keyword") ||
+            targetType.equals("constant.language");
+      
+      if (!isVariableType)
+         return -1;
+      
+      // Check to see if we're refactoring the name of an argument in a function call,
+      // e.g.
+      //
+      //    if (foo(apple = 1)) { ... }
+      //            ^^^^^
+      //
+      // If this is the case, then we want to search for all calls of this
+      // form and rename just that argument.
+      if (cursor.peekFwd(1).valueEquals("="))
+      {
+         String argName = cursor.currentValue();
+         TokenCursor clone = cursor.cloneCursor();
+         
+         while (clone.findOpeningBracket("(", false))
+         {
+            if (!clone.moveToPreviousToken())
+               break;
+            
+            String functionName = clone.currentValue();
+            return renameFunctionArgument(functionName, argName);
+            
+         }
+      }
+      
+      // Check to see if this is the name of a function argument. If so, we only want
+      // to rename within that scope.
+      Scope scope = editor_.getCurrentScope();
+      while (!scope.isTopLevel())
+      {
+         if (scope.isFunction())
+         {
+            boolean isCursorOnFunctionDefn =
+                  cursor.peek(1).isLeftAssign() &&
+                  cursor.peek(2).getValue().equals("function");
+            
+            if (!isCursorOnFunctionDefn)
+            {
+               ScopeFunction scopeFunction = (ScopeFunction) scope;
+               JsArrayString args = scopeFunction.getFunctionArgs();
+               for (int i = 0; i < args.length(); i++)
+                  if (args.get(i).equals(targetValue))
+                     return renameVariablesInScope(scope, targetValue, targetType);
+            }
+         }
+         scope = scope.getParentScope();
+      }
+      
+      // Otherwise, just rename the variable within the current scope.
+      // TODO: Do we need to look into parent scopes?
```

Comment:
Noticed a couple of TODOs in the diff. Let's either do these or remove the TODO comment entirely in favor of comment explaining limitations, caveats, etc.